iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -400,7 +428,14 @@ public override async Task<long> SendTwinPatchAsync(TwinCollection reportedPrope
             }
             finally
             {
-                _twinResponseCompletions.TryRemove(correlationId, out _);
+                if (_twinResponseCompletions.ContainsKey(correlationId))
+                {
+                    _twinResponseCompletions.TryRemove(correlationId, out _);
```

Comment:
You can combine these two calls into one like:
```
_ = _twinResponseCompletions.TryRemove(correlationId, out _)
```

TryRemove will remove it if it is present and will just return false if it was not present